larray/util/misc.py

```diff
@@ -627,6 +627,36 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def deprecate_kwarg(old_arg_name, new_arg_name, mapping=None, stacklevel=2):
+    if not isinstance(mapping, dict):
+        raise TypeError("mapping from old to new argument values must be dict!")
+    def _deprecate_kwarg(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            old_arg_value = kwargs.pop(old_arg_name, None)
+            if old_arg_value is not None:
+                if mapping is not None:
+                    new_arg_value = mapping.get(old_arg_value, old_arg_value)
+                    msg = "The {old_name}={old_val!r} keyword is deprecated, use {new_name}={new_val!r} instead"\
+                        .format(old_name=old_arg_name, old_val=old_arg_value, new_name=new_arg_name,
+                                new_val=new_arg_value)
+                else:
+                    new_arg_value = old_arg_value
+                    msg = "The '{old_name}' keyword is deprecated, use '{new_name}' instead"\
+                        .format(old_name=old_arg_name, new_name=new_arg_name)
+
+                warnings.warn(msg, FutureWarning, stacklevel=stacklevel)
+                if kwargs.get(new_arg_name, None) is not None:
```

Comment:
`if new_arg_name in kwargs:` ?

Reply:
This decorator comes directly from Pandas and may be a little old...

Reply:
tsss... We cannot just copy-paste code from Pandas without any notice...